rasa/nlu/components.py

```diff
@@ -91,6 +91,34 @@ def validate_requirements(component_names: List[Optional[Text]]) -> None:
         )
 
 
+def validate_component_keys(pipeline: List["Component"]) -> None:
+    """Validates that all keys for a component are valid.
+
+    Raises:
+        InvalidConfigError: If any component has a key specified that is not used
+            by the component class, it is likely a mistake in the pipeline
+
+    Args:
+        pipeline: The list of components in the piopeline
+    """
+    from rasa.nlu import registry
+
+    for component in pipeline:
+        component_name = component.get("name")
+        component_class = registry.get_component_class(component_name)
```

Comment:
Note that this may raise an exception. Maybe it would be better to run the validation within `_build_pipeline`?

Reply:
Or even better within `component.create_component`?

Reply:
that would also avoid having to do the check in two locations?

Reply:
mm I tried moving it into `component.create_component`, but in that case extra keys are present that aren't allowed for the child class, but are for the base class (e.g. `intent_classification` is present for `ResponseSelector`). So, it isn't only checking the user-provided config at that point. 

Re. putting it in `_build_pipeline` - you mean moving the logic inside the function instead of calling it from there? Which exception would that avoid?

Reply:
it would have the advantage that you don't have to load the components yourself in `_build_pipeline` (which can go wrong and then you'd need error handling).You'd already have the instantiated components in `_build_pipeline` which should simplify your validations.

Reply:
I moved it - does that work?